elements/lisk-p2p/test/functional/events/connection_create.ts

```diff
@@ -0,0 +1,169 @@
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { expect } from 'chai';
+import {
+	P2P,
+	EVENT_CONNECT_OUTBOUND,
+	EVENT_NEW_INBOUND_PEER,
+	EVENT_DISCOVERED_PEER,
+	EVENT_UPDATED_PEER_INFO,
+	EVENT_FAILED_TO_ADD_INBOUND_PEER,
+	INVALID_CONNECTION_SELF_REASON,
+	INCOMPATIBLE_NETWORK_REASON,
+	INCOMPATIBLE_PROTOCOL_VERSION_REASON,
+} from '../../../src/index';
+import { wait } from '../../utils/helpers';
+import {
+	createNetwork,
+	destroyNetwork,
+	SEED_PEER_IP,
+	NETWORK_START_PORT,
+} from '../../utils/network_setup';
+
+describe(`Connection Create`, () => {
+	describe(`Events`, () => {
+		let p2pNodeList: ReadonlyArray<P2P> = [];
+		const collectedEvents = new Map();
+
+		beforeEach(async () => {
+			// To capture all the initial events set network creation time to minimum 1 ms
+			p2pNodeList = await createNetwork({
+				networkSize: 2,
+				networkDiscoveryWaitTime: 1,
+			});
+			const firstNode = p2pNodeList[0];
+			const secondNode = p2pNodeList[1];
+
+			firstNode.on(EVENT_NEW_INBOUND_PEER, res => {
+				collectedEvents.set(EVENT_NEW_INBOUND_PEER, res);
+			});
+
+			secondNode.on(EVENT_CONNECT_OUTBOUND, res => {
+				collectedEvents.set(EVENT_CONNECT_OUTBOUND, res);
+			});
+			secondNode.on(EVENT_DISCOVERED_PEER, res => {
+				collectedEvents.set(EVENT_DISCOVERED_PEER, res);
+			});
+
+			secondNode.on(EVENT_UPDATED_PEER_INFO, res => {
+				collectedEvents.set(EVENT_UPDATED_PEER_INFO, res);
+			});
+
+			await wait(1000);
+		});
+
+		afterEach(async () => {
+			await destroyNetwork(p2pNodeList);
+		});
+
+		it(`should handle ${EVENT_NEW_INBOUND_PEER} event and payload`, async () => {
+			const secondNode = p2pNodeList[1];
+			const payload = collectedEvents.get(EVENT_NEW_INBOUND_PEER);
+
+			expect(payload)
+				.to.have.property('wsPort')
+				.which.equals(secondNode.nodeInfo.wsPort);
+			expect(payload).to.have.property('sharedState');
+		});
+
+		it(`should handle ${EVENT_CONNECT_OUTBOUND} event and payload`, async () => {
+			const firstNode = p2pNodeList[0];
+			const payload = collectedEvents.get(EVENT_CONNECT_OUTBOUND);
+
+			expect(payload)
+				.to.have.property('wsPort')
+				.which.equals(firstNode.nodeInfo.wsPort);
+			expect(payload).to.have.property('sharedState');
+		});
+
+		it(`should handle ${EVENT_UPDATED_PEER_INFO} event and payload`, async () => {
+			const firstNode = p2pNodeList[0];
+			const payload = collectedEvents.get(EVENT_UPDATED_PEER_INFO);
+
+			expect(payload)
+				.to.have.property('wsPort')
+				.which.equals(firstNode.nodeInfo.wsPort);
+			expect(payload).to.have.property('sharedState');
+
+			expect(collectedEvents.get(EVENT_UPDATED_PEER_INFO)).to.exist;
```

Comment:
Isn't it is unnecessary to check if `EVENT_UPDATED_PEER_INFO ` message exists or not as its already check above when we expect payload to have property?

Reply:
Also, is it possible to check if the updated peer was really updated in `peerBook` by checking in `getConnectedPeers` and `getDisconnectedPeers` for its state? this will help us to find potential errors revolving around peer update events